Objects/StrategyCardAutomator.ttslua

```diff
@@ -0,0 +1,1648 @@
+		broadcastToAll('Allocate command tokens: missing ' .. color .. ' Command Sheet', 'Red')
+		return
+	end
+
+	local pos = { x = -0.1, y = 1, z = -5 }
+	local dx = -0.4
+
+	-- Do the alloation.  Watch out for an empty bag!
+	for i = 1, count do
+		if commandTokensBag.getQuantity() == 0 then
+			_safeBroadcastToColor('You are out of Command Tokens. Only able to gain ' .. i .. ' Tokens. ', color, color)
+			break
+		end
+		local token = commandTokensBag.takeObject({
+			position = commandSheet.positionToWorld({
+				x = pos.x + (i * dx),
+				y = pos.y + 1 + (i * 0.25),
+				z = pos.z + dz
+			})
+		})
+		_strategyCardHelper._addUnallocatedCommandToken(token, color)
+	end
+
+	_safeBroadcastToColor('You have gained Command Tokens.', color, color)
+
+end
+
+------------------------------ secondary ---------------------------------------
+
+function _consumeStrategyToken(color)
+	local faction = _factionHelper.fromColor(color)
+	if not faction then
+		broadcastToAll('Consume Strategy tokens: no faction for ' .. color, 'Red')
+		return
+	end
+	log("consumeStrategyToken")
+	log(faction.name)
+
+	local nameToCommandTokensBag = {}
+
+	local SsruuActive = false
+	local JaeMirKanActive = false
+
+
+	-- find command token bag
+	-- also check if alternate command token payment method available.
+	-- if so, message the player and return true
+	for _, object in ipairs(getAllObjects()) do
+		local name = object.getName()
+		local guid = object.getGUID()
+		if object.tag == 'Bag' then
+			local tokenName = string.match(name, '^(.*) Command Tokens Bag')
+			if tokenName then
+				nameToCommandTokensBag[tokenName] = object
+			end
+		end
+
+		if (name == "Jae Mir Kan") then
+			JaeMirKanActive = true
+			if (faction.name == "The Mahact Gene-Sorcerers") and (not object.is_face_down) then
+				BroadcastToColor("Jae Mir Kan is available to you. No Token has been used.", color)
+				return true
+			end
+		elseif (faction.name == "The Yssaril Tribes") and (name == "Ssruu") and (not object.is_face_down) then
+			SsruuActive = true
+		end
+
+		if (name == "Scepter of Emelpar") and (not object.is_face_down) then
+			local zoneColor = _zoneHelper.zoneFromPosition(object.getPosition())
+			if zoneColor == color then
+				_safeBroadcastToColor("Scepter of Emelpar is available to you. No Token has been used.", color, color)
+				return true
+			end
+
+		end
+	end
+
+	if (SsruuActive and JaeMirKanActive) then
+		_safeBroadcastToColor("Ssruu is available to you. No Token has been used.", color, color)
+		return true
+	end
+
+
+	local commandTokensBag = nameToCommandTokensBag[faction.tokenName]
+	if not commandTokensBag then
+		broadcastToAll('Consume Strategy tokens: missing ' .. faction.tokenName .. ' Command Tokens Bag', 'Red')
+		return
+	end
+
+	local commandSheet = getObjectFromGUID(faction.commandSheetGuid)
+	if not commandSheet then
+		broadcastToAll('Consume Strategy tokens: missing ' .. color .. ' Command Sheet', 'Red')
+		return
+	end
+
+	local hitlist = Physics.cast({
```

Comment:
This works, I like to avoid casts where convenient so it could gather positions of flipped command tokens then look at where they are relative to the sheet via sheet.positionToLocal (take a look at the TI4 stats updater).